modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -5073,15 +5081,28 @@ private void _startWorkflowInstance(
 		boolean workflowEnabled = WorkflowThreadLocal.isEnabled();
 
 		try {
-			_skipModelListeners.set(skipModelListener);
-
-			ObjectActionThreadLocal.setSkipObjectActionExecution(true);
 			WorkflowThreadLocal.setEnabled(true);
 
-			WorkflowHandlerRegistryUtil.startWorkflowInstance(
-				objectEntry.getCompanyId(), objectEntry.getNonzeroGroupId(),
-				userId, objectDefinition.getClassName(),
-				objectEntry.getObjectEntryId(), objectEntry, serviceContext);
+			if (objectDefinition.isRootDescendantNode()) {
+				ObjectDefinition rootObjectDefinition =
+					_objectDefinitionPersistence.fetchByPrimaryKey(
+						objectDefinition.getRootObjectDefinitionId());
+
+				_startWorkflowInstance(
+					userId, rootObjectDefinition.getClassName(),
+					objectEntryPersistence.fetchByPrimaryKey(
+						objectEntry.getRootObjectEntryId()),
+					serviceContext);
+			}
+			else {
+				_skipModelListeners.set(skipModelListener);
```

Comment:
Is there any reason why this part of the code should not be executed when you start the workflow instance from a root object definition?

```
_skipModelListeners.set(skipModelListener);
ObjectActionThreadLocal.setSkipObjectActionExecution(true);
```

Reply:
@guilhermedcamacho during the meeting, we agreed that the actions for `on after update` should be executed when updating the status of the root object definition.